civihr_employee_portal/src/Helpers/WebformHelper.php

```diff
@@ -30,22 +45,25 @@ public static function getWebformComponentsByName($node, $name) {
    * Finds all webform submissions by a Drupal user.
    *
    * @param \stdClass $user
-   * @param string $name
+   * @param string $title
    *
    * @return array
    */
-  public static function getUserSubmissionsByName($user, $name) {
+  public static function getUserSubmissionsByTitle($user, $title) {
     $filters = ['uid' => $user->uid];
     module_load_include('inc', 'webform', 'includes/webform.submissions');
-    $submissions = webform_get_submissions($filters);
+    $allUserSubmissions = webform_get_submissions($filters);
+    $targetFormSubmissions = [];
+    $webform = self::findOneByTitle($title);
+    $webformId = $webform->nid;
 
-    foreach ($submissions as $submission) {
-      $title = property_exists($submission, 'title') ? $submission->title : NULL;
-      if ($title === $name) {
-        $submissions[] = $submission;
+    foreach ($allUserSubmissions as $submission) {
+      $nodeID = property_exists($submission, 'nid') ? $submission->nid : NULL;
+      if ($webformId === $nodeID) {
+        $targetFormSubmissions[] = $submission;
```

Comment:
Perhaps I'm slow today, but I can't see what is the difference here. Before we would only return the submissions with the same title as the given name, now we get the webform by title then return the submissions where the node ID is the same as the webform ID. Would you please give more details about this?

Reply:
I'm confused myself, when I was debugging this I couldn't find any submission which had the property 'title', must have just been a mistake by me from the beginning.